### ISSUE TITLE
Remove lingering Python 2 specific things

### ISSUE THREAD

**Issue by lesteve** [joblib/joblib#1079]
After https://github.com/joblib/joblib/pull/1018, there were things that were pushed as TODO python2_drop:

**Comment by OlehKSS** [comment]
As far as I see, several marked pieces of code should be removed because they are assumed to be obsolete, right? I can work on it provided nobody else has already started it.

**Comment by ogrisel** [comment]
Yes it's probably mostly about cleaning old, useless code. But one should be careful to avoid breaking things, possibly by adding new tests when we feel that the existing test suite is not enough.
